components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -0,0 +1,94 @@
+import { Project, Team } from "@gitpod/gitpod-protocol";
+import CheckBox from "../components/CheckBox";
+import { getGitpodService } from "../service/service";
+import { getCurrentTeam, TeamsContext } from "../teams/teams-context";
+import { PageWithSubMenu } from "../components/PageWithSubMenu";
+import PillLabel from "../components/PillLabel";
+
+export function getProjectSettingsMenu(project?: Project, team?: Team) {
+    const teamOrUserSlug = !!team ? 't/' + team.slug : 'projects';
+    return [
+        {
+            title: 'Configuration',
+            link: [`/${teamOrUserSlug}/${project?.slug || project?.name}/configure`],
+        },
+        {
+            title: 'Project',
+            link: [`/${teamOrUserSlug}/${project?.slug || project?.name}/settings`],
+        },
+    ];
+}
+
+export default function () {
+    const location = useLocation();
+    const { teams } = useContext(TeamsContext);
+    const team = getCurrentTeam(location, teams);
+    const match = useRouteMatch<{ team: string, resource: string }>("/(t/)?:team/:resource");
+    const projectSlug = match?.params?.resource;
+    const [ project, setProject ] = useState<Project | undefined>();
+
+    const [ isLoading, setIsLoading ] = useState<boolean>(true);
+    const [ isIncrementalPrebuildsEnabled, setIsIncrementalPrebuildsEnabled ] = useState<boolean>(false);
+
+    useEffect(() => {
+        if (!teams || !projectSlug) {
+            return;
+        }
+        (async () => {
+            const projects = (!!team
+                ? await getGitpodService().server.getTeamProjects(team.id)
+                : await getGitpodService().server.getUserProjects());
+
+            // Find project matching with slug, otherwise with name
+            const project = projectSlug && projects.find(p => p.slug ? p.slug === projectSlug : p.name === projectSlug);
+            if (!project) {
+                return;
+            }
+            setProject(project);
+        })();
+    }, [ projectSlug, team, teams ]);
+
+    useEffect(() => {
+        if (!project) {
+            return;
+        }
+        setIsLoading(false);
+        setIsIncrementalPrebuildsEnabled(!!project.settings?.useIncrementalPrebuilds);
+    }, [ project ]);
+
+    const toggleIncrementalPrebuilds = async () => {
+        if (!project) {
+            return;
+        }
+        setIsLoading(true);
+        try {
+            await getGitpodService().server.updateProjectPartial({
+                id: project.id,
+                settings: {
+                    useIncrementalPrebuilds: !isIncrementalPrebuildsEnabled,
+                }
+            });
+            setIsIncrementalPrebuildsEnabled(!isIncrementalPrebuildsEnabled);
+        } finally {
+            setIsLoading(false);
+        }
+    }
+
+    return <PageWithSubMenu subMenu={getProjectSettingsMenu(project, team)} title="Settings" subtitle={`Manage settings for your project ${project?.name}`}>
```

Comment:
**suggestion**: We already surface the project name in the top navigation menu. Shallow we remove it from here?

```suggestion
    return <PageWithSubMenu subMenu={getProjectSettingsMenu(project, team)} title="Settings" subtitle="Manage project settings and configuration">
```

Reply:
Good point, we already see the slug at the top (which should actually be the name, now that I think of it).

Will fix both. 😁